gaia-media-controls.js

```diff
@@ -0,0 +1,291 @@
+    if (this.mediaPlayerImpl) {
+      this.mediaPlayerImpl.unload();
+      this.mediaPlayerImpl = null;
+    }
+  },
+
+  /*
+   * Expose testing helper functions
+   */
+  enableComponentTesting() {
+    if (this._impl) {
+      this._impl.enableComponentTesting();
+
+      var componentTestingHelper = {
+        triggerEvent:
+          this._impl.triggerEvent.bind(this._impl),
+        getElement:
+          this._impl.getElement.bind(this._impl),
+        disableComponentTesting:
+          this._impl.disableComponentTesting.bind(this._impl)
+      };
+    }
+
+    return componentTestingHelper;
+  },
+
+  template: `
+
+  <style>
+
+  @font-face {
+  	font-family: "gaia-icons";
+  	src: url("fonts/gaia-icons.ttf") format("truetype");
+  	font-weight: 500;
+  	font-style: normal;
+  }
+
+  [data-icon]:before {
+  	font-family: "gaia-icons";
+  	content: attr(data-icon);
+  	display: inline-block;
+  	font-weight: 500;
+  	font-style: normal;
+  	text-decoration: inherit;
+  	text-transform: none;
+  	text-rendering: optimizeLegibility;
+  	font-size: 30px;
+  	-webkit-font-smoothing: antialiased;
+  }
+
+  .media-controls-container {
+    background-color: rgba(0,0,0, 0.85);
+    display: flex;
+    flex-direction: column;
+    justify-content: space-around;
+    align-items: stretch;
+    min-width: 30rem;
+  }
+
+  /* video bar -- duration, time slider, elapsed time */
+  .time-slider-bar {
+    display: flex;
+    flex-flow: row;
+    align-items: center;
+    font-size: 0;
+    border-bottom: 0.1rem solid rgba(255,255,255, 0.1);
+  }
+
+  /* 1. elapsed-text and duration-text have padding on left and right
+        to support ltr and rtl locales */
+  /* 2. The elapsed time and duration elements do not grow and shrink
+        via the flexbox. They are fixed width */
+  .elapsed-text, .duration-text {
+    color: #ffffff;
+    font-size: 1.4rem;
+    padding: 0 1.5rem; /* 1 */
+    flex-grow: 0;      /* 2 */
+    text-align: center;
+  }
+
+  /* 1. The slider element grows and shrinks via the flexbox */
+  .slider-wrapper {
+    flex-grow: 1;   /* 1 */
+    height: 4.2rem;
+  }
+
+  .progress {
+    position: relative;
+    pointer-events: none;
+    width: 0;
+  }
+
+  /* 1. Center elements vertically within time-slider; 'top: 50%' centers the
+   *    top of the element vertically; 'elapsed-time' is the first element
+   *    to be layed out, it is 0.3rem in height, therefore 'top:50%' would
+   *    position the center of the element 0.1rem below the middle: move
+   *    it up 0.1rem to center the middle of the element vertically.
+   *    'time-background' is layed out after 'elapsed-time' and is 0.1rem in
+   *    height. With 'elapsed-time' being 0.3rem in height, 'top:50%' would
+   *    position 'time-background' 0.3rem below the center vertically: move it
+   *    up 0.3rem to center it vertically.
+   *
+   * 2. Ensure the layering order of time background,
+        elapsed time, and play head.
+   */
```

Comment:
How does z-index interact with elements outside of the shadow root? It worries me to use constants here because of the possibility of interaction with elements outside of this component.  It seems to me that you ought to be able to get the layering correct just by putting the HTML elements in the proper order. Element later in the document should be on top of elements earlier in the document.